src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Created.cs

```diff
@@ -123,4 +124,87 @@ public static void FileSystemWatcher_Created_FileCreatedInNestedDirectory()
             }
         });
     }
+
+    // This can potentially fail on Windows, depending on where the test is run from, due to 
+    // the MAX_PATH limitation in Windows. When issue 645 is closed, this shouldn't be a problem
+    [Fact, ActiveIssue("645", PlatformID.Windows)]
+    public static void FileSystemWatcher_Created_DeepDirectoryStructure()
+    {
+        using (var dir = Utility.CreateTestDirectory())
+        using (var watcher = new FileSystemWatcher())
+        {
+            AutoResetEvent eventOccured = Utility.WatchForEvents(watcher, WatcherChangeTypes.Created);
+
+            // put everything in our own directory to avoid collisions
+            watcher.Path = Path.GetFullPath(dir.Path);
+            watcher.Filter = "*.*";
+            watcher.IncludeSubdirectories = true;
+            watcher.EnableRaisingEvents = true;
+
+            // List of created directories
+            List<TemporaryTestDirectory> lst = new List<TemporaryTestDirectory>();
+
+            // Priming directory
+            TemporaryTestDirectory priming = new TemporaryTestDirectory(Path.Combine(dir.Path, "dir"));
+            lst.Add(priming);
+            Utility.ExpectEvent(eventOccured, "priming create");
+
+            // Create a deep directory structure and expect things to work
+            for (int i = 1; i < 20; i++)
+            {
+                lst.Add(new TemporaryTestDirectory(Path.Combine(lst[i - 1].Path, String.Format("dir{0}", i))));
+                Utility.ExpectEvent(eventOccured, lst[i].Path + " create");
+            }
+
+            // Put a file at the very bottom and expect it to raise an event
+            using (var file = new TemporaryTestFile(Path.Combine(lst[lst.Count - 1].Path, "temp file")))
+            {
+                Utility.ExpectEvent(eventOccured, "temp file create");
```

Comment:
Are we validating anywhere for any/all of these tests that the path information included in the raised events is correct?  Since we're manually constructing those paths in the Linux implementation in particular, that'd be a good thing to verify.